### PR TITLE
[ip6-address/address-resolver/commissioner] use ExtAddress helper methods

### PR DESCRIPTION
**Pull request by abtink**

### CODE REVIEW

**Comment by jwhui** [comment]
There are other instances in core (e.g. commissioner, address resolver) where we could make use of `ToggleLocal()` and replace `memcpy` with `=`. Shall we update those too?

**Comment by abtink** [comment]
Sounds good! Will take care of them (from same PR).

**Comment by abtink** [comment]
Submitted https://github.com/openthread/openthread/pull/4183 for the address-resolver and updated https://github.com/openthread/openthread/pull/4180 for commissioner.

File: src/core/net/ip6_address.cpp [hunk 1]
```diff
@@ -169,13 +169,15 @@ void Address::SetIid(const uint8_t *aIid)
 
 void Address::SetIid(const Mac::ExtAddress &aExtAddress)
 {
-    memcpy(mFields.m8 + kInterfaceIdentifierOffset, aExtAddress.m8, kInterfaceIdentifierSize);
-    mFields.m8[kInterfaceIdentifierOffset] ^= 0x02;
+    Mac::ExtAddress *iid = reinterpret_cast<Mac::ExtAddress *>(mFields.m8 + kInterfaceIdentifierOffset);
```
Comment (jwhui) [inline src/core/net/ip6_address.cpp]: nit: declare `iid` as reference rather than pointer?

**Comment by abtink** [comment]
We are getting a weird error on travis
```
/home/travis/build/openthread/openthread/examples/../src/core/thread/thread_tlvs.hpp:308:77: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
         Mac::ExtAddress &iid = *reinterpret_cast<Mac::ExtAddress *>(&mIid[0]);
```
`ExtAddress` is a packed type so should not cause issue pointing to any memory address. We use similar code elsewhere with no issue.

**Pull request merged**